docs/plugins.md

```diff
@@ -35,6 +35,24 @@ spec:
 
 ## Considerations
 
+### Security
+
+Plugins are a powerful way to augment Argo Workflows' capabilities. Plugins also introduce code which may contain 
+vulnerabilities.
+
+Before enabling plugins, consider:
+* Who should be able to install plugins? How will this be enforced (e.g. RBAC rules for creating/modifying ConfigMaps)?
+
+When developing plugins, consider:
```

Comment:
maybe we should provide some way to scan plugins?

Reply:
Scan for common weaknesses (e.g. SQL injection) or something different?